### PR TITLE
Refactor frame handling

### PR DESCRIPTION
**Pull request by shaleh**
Break the core logic of `framep` out into a separate function. Use
this function internally since we already have `LispFrameRef` objects.

This allows `frame_live_p` to not duplicate the frame checks again in
`framep`.

`window-system` should just return a `LispObject`. There is no reason
to convert to a Symbol and immediately turn it back into a
`LispObject`.

### CODE REVIEW

File: rust_src/src/frames.rs [hunk 1]
```diff
@@ -225,14 +227,19 @@ pub fn framep(object: LispObject) -> LispObject {
 /// use `display-graphic-p' or any of the other `display-*-p'
 /// predicates which report frame's specific UI-related capabilities.
 #[lisp_fn(min = "0")]
-pub fn window_system(frame: Option<LispFrameRef>) -> Option<LispSymbolRef> {
-    let f = frame.map_or(selected_frame(), |f| f.as_lisp_obj());
-    let window_system = framep(f);
+pub fn window_system(frame: Option<LispFrameRef>) -> LispObject {
+    let frame = frame.unwrap_or_else(|| selected_frame().as_frame_or_error());
```
Comment (brotzeit) [inline rust_src/src/frames.rs]: Do we really need to check if selected_frame is a frame ? If we do I think an assertion would be better.

File: rust_src/src/frames.rs [hunk 2]
```diff
@@ -225,14 +227,19 @@ pub fn framep(object: LispObject) -> LispObject {
 /// use `display-graphic-p' or any of the other `display-*-p'
 /// predicates which report frame's specific UI-related capabilities.
 #[lisp_fn(min = "0")]
-pub fn window_system(frame: Option<LispFrameRef>) -> Option<LispSymbolRef> {
-    let f = frame.map_or(selected_frame(), |f| f.as_lisp_obj());
-    let window_system = framep(f);
+pub fn window_system(frame: Option<LispFrameRef>) -> LispObject {
```
Comment (brotzeit) [inline rust_src/src/frames.rs]: I've thought that we changed our return types and now use more explicit types. You didn't complain about this in the last PRs ?

**Comment by brotzeit** [comment]
> window-system should just return a LispObject. There is no reason
> to convert to a Symbol and immediately turn it back into a
> LispObject.

I have asked myself the same thing. But I thought it's about indicating the return type.

**Comment by shaleh** [comment]
The purpose of the recent changes is to avoid having the developer juggle types. If a function is always working on a Window the developer should not need to switch back and forth between the LispObject and the Window. If that Window is returned then it should be the return type.

In the case of this function the returned object is always used as a LispObject in the function so it might as well stay one and be returned appropriately.

Basically, write the function using the most natural types possible. That may be LispObject or that may be bool or something else. Then write the function definition to match. This means some functions will take 2 LispObjects and return another one. Others will take a LispFrameRef and a bool and return a LispWindowRef.
